lib/elastic_apm/sinatra.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  # Module for starting the ElasticAPM agent and hooking into Sinatra.
+  module Sinatra
+    extend self
+    # Start the ElasticAPM agent and hook into Sinatra.
+    #
+    # @param app [Sinatra::Base] A Sinatra app.
+    # @param config [Config, Hash] An instance of Config or a Hash config.
+    # @return [true, nil] true if the agent was started, nil otherwise.
+    def start(app, config)
+      config = Config.new(config) unless config.is_a?(Config)
+      set_config(app, config)
+
+      ElasticAPM.start(config)
+      ElasticAPM.running?
+    rescue StandardError => e
+      config.logger.error format('Failed to start: %s', e.message)
+      config.logger.debug "Backtrace:\n" + e.backtrace.join("\n")
+    end
+
+    private
+
+    def set_config(app, config)
```

Comment:
I don't think this method name describes what's going on. How about `set_app` or `configure_app` or something similar?